packages/lex_service/src/lexicon/resolvers/lexicon.resolver.ts

```diff
@@ -16,6 +17,20 @@ export class LexiconResolver {
     return this.lexiconService.create(lexicon);
   }
 
+  @Mutation(() => Lexicon)
+  @UseGuards(AuthGuard)
+  async lexiconUpdate(
+  @Args('id') id: string,
+  @Args('name', { nullable: true }) name?: string,
```

Comment:
Instead of listing out the fields that can be changed, I recommend leveraging NestJS's built in GraphQL operations


```
import { Bundle } from '../bundles.model';
import { ID, InputType, OmitType, PartialType, Field } from '@nestjs/graphql';

@InputType()
export class BundleChange extends PartialType(OmitType(Bundle, ['id', 'services'] as const), InputType) {
  @Field(() => [ID], { nullable: true })
  services: string[];
}
```
I would recommend creating a new dto for updating a lexicon which omits the ID and is then a partial type. The benefit of this approach is you then don't have to list out all the fields. If we make a chance to the `Lexicon` model, the update dto will also reflect those changes.

Reply:
I'm a little unsure about this comment, but I believe I have worked around it. You will see it in the updated pull request.